composer/trainer/trainer_hparams.py

```diff
@@ -222,6 +221,10 @@ def validate(self):
                 f"Eval batch size ({self.eval_batch_size}) not divisible by the total number of processes ({world_size})."
             )
 
+        # There should be at least one structure for validation - evaluators or ordinary validation dataset
+        if self.val_dataset is None and self.evaluators is None:
```

Comment:
Sometimes (e.g. in tests) I'd like to do a run without validation. I think it'd make sense to raise a warning but not error out.
In addition, would it make sense to remove the `val_dataset` hparam since now that should be specified in `evaluators`?